core/embed/rust/src/storagedevice/storage_device.rs

```diff
@@ -0,0 +1,690 @@
+        let delay = AUTOLOCK_DELAY_MS.get().unwrap_or(AUTOLOCK_DELAY_DEFAULT);
+        _normalize_autolock_delay(delay).try_into()
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_autolock_delay_ms(delay_ms: Obj) -> Obj {
+    let block = || AUTOLOCK_DELAY_MS.set(delay_ms.try_into()?)?.try_into();
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_flags() -> Obj {
+    let block = || FLAGS.get().unwrap_or(0).try_into();
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_flags(flags: Obj) -> Obj {
+    let block = || {
+        let flags: u32 = flags.try_into()?;
+
+        let old_flags = FLAGS.get().unwrap_or(0);
+
+        // Not deleting old flags
+        let new_flags = flags | old_flags;
+        FLAGS.set(new_flags)?.try_into()
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_safety_check_level() -> Obj {
+    let block = || {
+        let level = SAFETY_CHECK_LEVEL
+            .get()
+            .unwrap_or(DEFAULT_SAFETY_CHECK_LEVEL);
+
+        let level = if !SAFETY_CHECK_LEVELS.contains(&level) {
+            DEFAULT_SAFETY_CHECK_LEVEL
+        } else {
+            level
+        };
+
+        Ok(level.into())
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_safety_check_level(level: Obj) -> Obj {
+    let block = || {
+        let level = level.try_into()?;
+
+        if !SAFETY_CHECK_LEVELS.contains(&level) {
+            Err(Error::ValueError(cstr!("Not valid safety level")))
+        } else {
+            SAFETY_CHECK_LEVEL.set(level)?.try_into()
+        }
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_sd_salt_auth_key() -> Obj {
+    let block = || {
+        SD_SALT_AUTH_KEY
+            .get()
+            .map_or(Ok(Obj::const_none()), |x| (&x as &[u8]).try_into())
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_sd_salt_auth_key(auth_key: Obj) -> Obj {
+    let block = || {
+        let auth_key: Buffer = auth_key.try_into()?;
+
+        if auth_key.is_empty() {
+            SD_SALT_AUTH_KEY.delete()?.try_into()
+        } else {
+            SD_SALT_AUTH_KEY.set(auth_key.try_into()?)?.try_into()
+        }
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_next_u2f_counter() -> Obj {
+    let block = || {
+        let key = helpers::get_appkey_u2f(APP_DEVICE, U2F_COUNTER, true);
+        storage::get_next_counter(key)?.try_into()
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_set_u2f_counter(count: Obj) -> Obj {
+    let block = || {
+        let key = helpers::get_appkey_u2f(APP_DEVICE, U2F_COUNTER, true);
+        storage::set_counter(key, count.try_into()?)?.try_into()
+    };
+    unsafe { util::try_or_raise(block) }
+}
+
+extern "C" fn storagedevice_get_private_u2f_counter() -> Obj {
```

Comment:
leftover?

Reply:
This and the one below are actually used, these are some strange cases being called in `storage/__init__.py::_migrate_from_version_01()`. Not sure why exactly they are needed, and even how to test it - as the migration could be only possible on the physical device